### PR TITLE
Fixes #4684 - Common Marketing Banner

### LINKED ISSUE #4684: [FxA] Implement common marketing view (also in header)
Looking at the [new spec](
https://mozilla.github.io/testpilot-assets/Firefox_Screenshots/NEW_FxA_Integration/), there's a common component used as an advertising banner across all pages. In some cases it has a little 🥕 pointing to the sign in button. 

Refer to Shot View, My Shots and Home Page specs

It should accept arbitrary text and inline elements as children.

### PR DESCRIPTION
**Pull request by punamdahiya**

### CODE REVIEW

File: locales/en-US/server.ftl [hunk 1]
```diff
@@ -9,6 +9,7 @@ gNoShots
 gScreenshotsDescription = Screenshots made simple. Take, save, and share screenshots without leaving Firefox.
 gSettings = Settings
 gSignIn = Sign In
+gBannerMessage = Sign in or sign up to sync shots across devices, save your favorite shots forever.
```
Comment (johngruen) [inline locales/en-US/server.ftl]: Change this to: 

Sign in or sign up to sync shots across devices and save your favorite shots forever.

**Comment by johngruen** [comment]
@punamdahiya this looks great...little string change and a little rebase, but R+ from me

**Comment by punamdahiya** [comment]
Updated and rebased. Thanks!

**Comment by 6a68** [comment]
Looking now. Not sure what's up with the test failures, I think I retriggered them.

**Comment by punamdahiya** [comment]
@6a68 My first iteration of this PR designed the way you have described above
a) content inline elements in the banner - upsell (for non-Firefox users) vs default global message (FxA specific)
b) pass booleans shouldGetFirefox, isOwner, and hasFxA as props to AdBanner and conditionally display above content

This has the benefit of AdBanner encapsulating logic around messages shown in banner and will work for our current use case where we have one global message for authenticated state and only one upsell message shown only on shot page.

However this limits us if in future we want to show different messages on different pages based off different conditional logic.

Looking [at the requirement](https://github.com/mozilla-services/screenshots/issues/4684#issue-343123499) in the #4684, “Marketing banner should accept arbitrary text and inline elements as children”

I changed design to have AdBanner flexibility of showing dynamic content  passed by specific page headers.  Let’s weigh in pros and cons of both approaches in standup tomorrow and can update accordingly. Thanks!

**Comment by punamdahiya** [comment]
@6a68 As discussed, updated AdBanner component to render banner content. Thanks!

File: server/src/config.js [hunk 2]
```diff
@@ -319,6 +319,13 @@ const conf = convict({
     env: "DISABLE_SEARCH",
     arg: "disable-search",
   },
+  disableBanner: {
+    doc: "If true, then hide the banner",
+    format: Boolean,
+    default: false,
+    env: "DISABLE_BANNER",
+    arg: "disable-banner",
+  },
```
[the remark this answers is not in the document]
Comment (punamdahiya) [reply server/src/config.js]: Done, this simplified PR a lot!

File: server/src/header.js [hunk 3]
```diff
@@ -9,13 +10,21 @@ exports.Header = function Header(props) {
       </Localized>
     </div> : null;
 
-  return <header className="header-panel default-color-scheme">
+  return <div className="header-container">
```
[the remark this answers is not in the document]
Comment (punamdahiya) [reply server/src/header.js]: +1, styles diff should be easier to review with this change

File: server/src/ad-banner.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+          <span>Screenshots made simple. Take, save and share screenshots without leaving Firefox.</span>
+        </Localized>
+        <Localized id="shotPageUpsellFirefox">
+          <a className="upsellLink"
+            href="https://www.mozilla.org/firefox/new/?utm_source=screenshots.firefox.com&utm_medium=referral&utm_campaign=screenshots-acquisition&utm_content=from-shot"
+            onClick={ this.clickedInstallFirefox.bind(this) }>Get Firefox now</a>
+        </Localized></div>;
+    } else if (this.props.isOwner && !this.props.hasFxa) {
+      bannerContent = <Localized id="gBannerMessage">
+          <span>
```
[the remark this answers is not in the document]
Comment (punamdahiya) [reply server/src/ad-banner.js]: update to use <p> by overriding it default display block style

**Comment by punamdahiya** [comment]
@6a68 Updated PR with feedback, thanks!

**Comment by punamdahiya** [comment]
Thanks @6a68, I should update firefox logo here https://github.com/mozilla-services/screenshots/pull/4910/commits/fda8dbe04170306cf619909422a612e0b428ff2e#diff-7602a15993b5c930593212bcdac6ba61R179 with updated one from Mark before landing. WIll do that now

**Comment by 6a68** [comment]
Looks like the tests are timing out again. I'm just going to land this.

**Pull request merged**